phoenix-core-client/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -190,8 +192,20 @@ private void verifySCN() throws SQLException {
         if (scn == null) {
             return;
         }
-        long maxLookBackAgeInMillis =
-            BaseScannerRegionObserverConstants.getMaxLookbackInMillis(conn.getQueryServices().getConfiguration());
+        ColumnResolver resolver = FromCompiler.getResolverForQuery(select, conn);
```

Comment:
I do not think we need to get a new ColumnResolver here. Can we use the column resolver of the QueryCompiler object. Please note FromCompiler.getResolverForQuery() makes an RPC call to the syscat region.